core/beacon/types.go

```diff
@@ -72,6 +75,18 @@ type executableDataMarshaling struct {
 	Transactions  []hexutil.Bytes
 }
 
+//go:generate go run github.com/fjl/gencodec -type ExecutableDataV2 -field-override executableDataV2Marshaling -out gen_edv2.go
+
+type ExecutableDataV2 struct {
```

Comment:
Should probably rename this as ExecutableDataV2 means actually the payload information.